serde-generate/runtime/typescript/serde/binarySerializer.ts

```diff
@@ -17,7 +17,7 @@ export abstract class BinarySerializer implements Serializer {
     private buffer: ArrayBuffer;
     private offset: number;
 
-    protected constructor() {
```

Comment:
Not a specialist of JS/TS but I suspect a protected constructor could make sense for an abstract class.

Reply:
If the class is abstract the constructor cannot be called. Having a protected constructor only forces that the class overrides it just for the sake of making it not protected again, which introduces possible refactor pitfalls (if you add an argument to the constructor, remember to go change all the other constructors to mirror it - again, their only purpose is to set it back to public).